lua/telescope/pickers.lua

```diff
@@ -1544,11 +1544,17 @@ function pickers.on_close_prompt(prompt_bufnr)
           picker.manager = EntryManager:new(picker.max_results, picker.entry_adder, picker.stats)
         end
       end
-      picker.default_text = picker:_get_prompt()
+      local curr_prompt = picker:_get_prompt()
+      picker.default_text = curr_prompt
       picker.cache_picker.selection_row = picker._selection_row
-      picker.cache_picker.cached_prompt = picker:_get_prompt()
+
+      -- only cache if prompt is not empty
+      if not (curr_prompt == nil or curr_prompt == "") then
+        picker.cache_picker.cached_prompt = curr_prompt
+        table.insert(cached_pickers, 1, picker)
+      end
+
       picker.cache_picker.is_cached = true
```

Comment:
Should these `picker.*` values only set inside the if statement?

Reply:
Good catch! Done :) Also tested locally, it works as expected. Could you please help approve workflows for this PR so that I can check if there is any workflow failing? Thank you!